src/init.cpp

```diff
@@ -1314,30 +1314,33 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
         }
     }
 
-    for (const std::string port_option : {
-        "-i2psam",
-        "-onion",
-        "-proxy",
-        "-rpcbind",
-        "-torcontrol",
-        "-whitebind",
-        "-zmqpubhashblock",
-        "-zmqpubhashtx",
-        "-zmqpubrawblock",
-        "-zmqpubrawtx",
-        "-zmqpubsequence",
+    for (const auto &port_option : std::vector<std::pair<std::string, bool>>{
+        // arg name            UNIX socket support
+        {"-i2psam",                 false},
+        {"-onion",                  true},
+        {"-proxy",                  true},
+        {"-rpcbind",                false},
+        {"-torcontrol",             false},
+        {"-whitebind",              false},
+        {"-zmqpubhashblock",        true},
+        {"-zmqpubhashtx",           true},
+        {"-zmqpubrawblock",         true},
+        {"-zmqpubrawtx",            true},
+        {"-zmqpubsequence",         true}
     }) {
-        for (const std::string& socket_addr : args.GetArgs(port_option)) {
+        const std::string arg{port_option.first};
+        const bool unix{port_option.second};
+        for (const std::string& socket_addr : args.GetArgs(arg)) {
             std::string host_out;
             uint16_t port_out{0};
             if (!SplitHostPort(socket_addr, port_out, host_out)) {
 #if HAVE_SOCKADDR_UN
-                // Allow unix domain sockets for -proxy and -onion e.g. unix:/some/file/path
-                if ((port_option != "-proxy" && port_option != "-onion") || socket_addr.find(ADDR_PREFIX_UNIX) != 0) {
-                    return InitError(InvalidPortErrMsg(port_option, socket_addr));
+                // Allow unix domain sockets for some options e.g. unix:/some/file/path
+                if (!unix || socket_addr.find(ADDR_PREFIX_UNIX) != 0) {
```

Comment:
If adjusted to:
```c
if (!unix || (socket_addr.find(ADDR_PREFIX_UNIX) != 0 && socket_addr.find(ADDR_PREFIX_IPC) != 0))
```

it seems like support for both forms (`unix:` and `ipc://`) might be achievable. 

If I'm not missing something, 
https://github.com/bitcoin/bitcoin/blob/7b4c7bdb35eda125781d14535a69dc6b9898069d/src/zmq/zmqnotificationinterface.cpp#L63
would allow `ipc://` to pass through (while also normalizing `unix:` to `ipc://`).

Was there a reason/rationale to use only `unix:` rather than both forms?  I would guess that folks familiar with ZMQ would be used to using `ipc://`, so it may be more straightforward to them to include support for that form.

I'm ok with either approach (supporting only `unix:` or supporting both `unix:` and `ipc://`).

Reply:
Heh yeah actually the branch used to be exactly like this but after #27375 got revised I simplified it. I'm open to discuss options, we could support any combination of `unix:` `unix://` `ipc:` `ipc://` or use a regex to cover them all. The motivation for keeping it simple isn't super strong but goes like this:

- Tor and libzmq only support one prefix each, and not even the same one. Meanwhile LND supports another set of prefixes.
- We would also need to update `IsUnixSocketPath()` to be more flexible 
- We'd also need a new function `GetUnixPathFromArg()` or something to replace lines like this that only expect one prefix:

https://github.com/bitcoin/bitcoin/blob/015ac13dcc964a31ef06dfdb565f88f901607f0e/src/netbase.cpp#L649

Reply:
Ah, thanks for the explanation.  TLDR:  After some thought, my vote is support only the `unix:` prefix.

Some thoughts for posterity.  Seems like the following are (generally) the case:
- Supporting a straightforward and uniform way to indicate usage of UNIX domain sockets makes sense (and can be less confusing to the user).  Currently Core supports unix domain sockets with `-proxy` and `-onion` (thanks to PR 27375!), and the goal is to add it for more areas, so whatever prefix is chosen should be reasonable for each of the options that can use it.
  - In the long run, support for UNIX domain sockets would be added for ZMQ (this PR), with I2P SAM proxy, on Windows, and in the GUI.  I'm not seeing use of a single prefix `unix:` being an issue for any of these applications (even in Windows, where the socket type is still called `AF_UNIX` AFAIK).
  - Users typically experience confusion when there are many possible ways to accomplish the goal (e.g. "Wait, which prefix should I use?")
  - There would be a ripple effect for guides written in the future (e.g. some tutorials/guides would state to use one prefix, and some would state to use another).  This could add confusion for users.
- Some prefixes make more sense for some options than others.  `unix:` seems to be the most straightforward and universally applicable.  `ipc://` could possibly make sense for ZMQ options since ZMQ uses the prefix, but it wouldn't make much sense for `-proxy` with Tor, so may even be a little confusing to support it there.
- Supporting multiple indicators (e.g. `unix:`, `ipc://`, etc.) simultaneously means more code and more complexity to maintain.  The added value for users accustomed to using `ipc://` with non-Bitcoin ZMQ application seems minimal compared to the effort for code maintenance.  These users could simply start using `unix:` for Bitcoin application.
- More code and more complexity typically means more bugs, vulns, more review burden, etc.  If there was a massive value gain it might be worth the effort, but doesn't like that's the case here.